apps/examples/st_things/msghandling/sample_msghandling_user.c

```diff
@@ -174,10 +202,41 @@ bool handle_set_request_on_resource_wwstfilterstateresuri(st_things_set_request_
 {
 	char *machinestates;
 	if (req_msg->rep->get_str_value(req_msg->rep, PROP_MACHINESTATES, &machinestates)) {
-		strncpy(g_machine_state, machinestates, 10);
+		strncpy(g_machine_state, machinestates, MAX_STR_LEN);
 		resp_rep->set_str_value(resp_rep, PROP_MACHINESTATES, machinestates);
 		st_things_notify_observers(req_msg->resource_uri);
 	}
 
 	return true;
 }
+
+bool handle_get_request_on_resource_wwstrobotcleanerturbomoderesuri(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+	if (req_msg->has_property_key(req_msg, PROP_MODES)) {
+		resp_rep->set_str_array_value(resp_rep, PROP_MODES, (const char **)g_modes, g_mode_cnt);
+	}
+	if (req_msg->has_property_key(req_msg, PROP_SUPPORTEDMODES)) {
+		resp_rep->set_str_array_value(resp_rep, PROP_SUPPORTEDMODES, (const char **)g_supported_modes, MAX_SUPPORTED_MODES);
+	}
+
+	return true;
+}
+
+bool handle_set_request_on_resource_wwstrobotcleanerturbomoderesuri(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+	char **req_str_array;
+	size_t array_size;
+	int idx;
+
+	if (req_msg->rep->get_str_array_value(req_msg->rep, PROP_MODES, &req_str_array, &array_size)) {
+		for (idx = 0; idx < g_mode_cnt; ++idx) {
+			free(g_modes[idx]);
```

Comment:
I saw only two allocation for g_modes.
One is for g_modes, another is g_modes[0].
Why do you free g_modes with g_mode_cnt?

Reply:
In the init_modes(), g_modes  is set char* array and got an element.
And then g_modes, g_mode_cnt are updated when it receivces post reqeust from the client.
So the setted datas must be freed.